Makefile

```diff
@@ -26,6 +26,10 @@ endif
 # Set Shell to bash, otherwise some targets fail with dash/zsh etc.
 SHELL := /bin/bash
 
+# ensure npm is accessible for autorest
+NODE_PATH = $(shell ./find-node-or-install)
+PATH := $(NODE_PATH):$(shell echo $$PATH)
+
```

Comment:
We don't need to ensure that node itself is available. We can make this a prerequisite in the Readme so that it's in the Developer's PATH if one wants to update the REST package, but I would not make it part of every single build, since this operator's development requirement is primarily Go runtime and not node too

Reply:
I especially wanted to be sure that it's available on the build runner. But it's true that the client doesn't need to be regenerated unless the swagger file changed, so it would be slimmer not to have the Makefile always ensure node is available.

Reply:
I have moved this part into a separate target that is only used by the autorest target.

Reply:
If it's supposed to be generated in CI, we would add a "setup-npm" GH action, just like we have with "setup-go". It's easier to reuse something that is already well-maintained :) In our case I would leave it to the developers how they want to install npm, just like we don't need to install Go for them.